### ISSUE TITLE
Evaluate PN and PA

### ISSUE THREAD

**Issue by elboyran** [dianna-ai/dianna-exploration#29]
Evaluate the PatternNet and PatternAttribution methods based on the evaluation criteria from issue #9 and fill in the results in the table created in issue #23.

- [Paper](https://arxiv.org/pdf/1705.05598.pdf)

Note: Paper claims other methods like LRP are wrong for simple linear models, while their methods are correct! BUT Explainer needs training. Implementations exist, e.g. [iNNvestigate](https://github.com/albermax/innvestigate)

**Comment by elboyran** [comment]
To be disscusses in the standup:
According to the PN & PA paper's taxonomy of methods those are of 3 types:

> the different types of explanation methods which can be divided into function,signal and attribution visualizations.  These three groups all present different information about the network and complement each other.

(observed data x = s + d, s- signal, d- distractor, s = a*y, y is what we try to estimate, the filter (with weights w) filters out the distractor )

- function (e.g. gradient, smooth grad). 

> Explaining the function in input space corresponds to describing the operations the model uses to extract y from x.

- signal (e.g. PN, DeconvNet, Guided backprop)

> The signal s detected by the neural net-work is the component of the data that caused the networks activations

- attribution (e.g. LRP, DTD, DeepLIFT, PA)

> how much the signal dimensions contribute to the output through the layers

**1. How do those correspond to the F3 criteria (Explanation target) - data, model, prediction?
2. Shall we limit ourselves only to Attribution methods (e.g. predictions??), so no need to evaluate DeconvNet etc. or keep them for alternative view?**

For illustration of the outputs of the 3 types of methods refer to [iNNvestigate README](https://github.com/albermax/innvestigate) tittle figure (the same taxonomy is given later on the same page).

**Comment by elboyran** [comment]
After a short discussion, we concluded that the taxonomy in the PN&PA paper roughly corresponds to the F3 criteria:
 

- function <--> model
- signal <--> data
- attribution <--> prediction

We have made the choice for F3 to focus on predictions and possibly model, not data, but we could consider several attribution methods and 1 for the group "function" (e.g. smooth grad) and 1 for "signal" (e.g. Pattern Net)
